scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -987,6 +989,62 @@ def publish_visibility(self) -> PublishVisibility:
         return PublishVisibility(visibility) if visibility else None
 
 
+class GenericAssetConfig(Config):
+    """Generic Asset Config class.
+
+    Example:
+        # Remote storage path for asset data
+        remote_path:
+            type: azureblob
+            storage_name: my_storage
+            container_name: my_container
+            container_path: foo/bar
+        # Local folder path to copy remote files into
+        local_path: ./prompt
+    """
+
+    def __init__(self, file_name: Path):
+        """Initialize object for the Generic Asset Properties extracted from storage.yaml.
+
+        Args:
+            file_name (Path): Storage config file to load and validate.
+        """
+        super().__init__(file_name)
+        self._path = None
+        self._validate()
+
+    def _validate(self):
+        """Validate the yaml file."""
+        Config._validate_exists('generic_asset.path', self.remote_path)
+        Config._validate_enum('generic_asset.path.type', self.remote_path.type.value, PathType, True)
+        Config._validate_exists('generic_asset.local_path', self.local_path)
+
+    @property
+    def remote_path(self) -> AssetPath:
+        """Remote storage path (Azure Blob is the only supported type)."""
+        if self._path:
+            return self._path
+        path = self._yaml.get('remote_path', {})

```

Comment:
Ideally this should follow the convention used by ModelConfig, where there's a `path` property and then `type` and other  properties are underneath it. Then you wouldn't have a `local_path` property for GenericAssetConfig.

Reply:
I refactored to just have a single 'path' property and remove the part that copies into local_path.